src/mon/MDSMonitor.cc

```diff
@@ -767,6 +768,19 @@ bool MDSMonitor::preprocess_command(MMonCommand *m)
       if (p != &mdsmap)
 	delete p;
     }
+  } else if (prefix == "mds metadata") {
+    mds_gid_t gid = MDS_GID_NONE;
+    if (!cmd_getval(g_ceph_context, cmdmap, "gid", gid)) {
+      ss << "error parsing 'gid' value '"
+         << cmd_vartype_stringify(cmdmap["gid"]) << "'";
+      return -EINVAL;
+    }
+    if (!f)
+      f.reset(Formatter::create("json-pretty"));
+    f->open_object_section("mon_metadata");
+    r = get_mds_metadata(gid, f.get(), ds);
```

Comment:
we mostly use "get" for methods that return data structures, and "dump" for methods that write out to a Formatter, so this should be dump_mds_metadata